stdlib/src/testing/testing.mojo

```diff
@@ -41,21 +41,32 @@ fn _isclose(
     rtol: Scalar[a.type],
     equal_nan: Bool,
 ) -> SIMD[DType.bool, a.size]:
+    constrained[
+        a.type.is_bool() or a.type.is_integral() or a.type.is_floating_point(),
+        "input type must be boolean, integral, or floating-point",
+    ]()
+
     @parameter
     if a.type.is_bool() or a.type.is_integral():
         return a == b
+    else:
```

Comment:
I think you can unindent the whole block and simply remove `else:`.

Reply:
Thank you so much for your review!

I can't do that since `DType.bool` isn't a numeric type. For example, the operation `a - b` isn't defined when `a` and `b` are boolean vectors.

Reply:
Oh, I meant using the [guard pattern](https://en.wikipedia.org/wiki/Guard_(computer_science)), which helps keeping the main logic as the function body.
```mojo
@parameter
if cond:
  ...
else:
  ...

# to 
@parameter
if cond:
  ...
...
```

Reply:
I understood what you meant. Have you tried making this change? Here it results in the following error:

```
FAIL: Mojo Standard Library :: testing/test_assertion.mojo (1 of 83)
******************** TEST 'Mojo Standard Library :: testing/test_assertion.mojo' FAILED ********************
Exit Code: 1

Command Output (stderr):
--
RUN: at line 13: mojo -debug-level full /home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo
+ mojo -debug-level full /home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:136:4: error: function instantiation failed
fn __mojo_main_prototype(
   ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:136:4: note:   call expansion failed - no concrete specializations
fn __mojo_main_prototype(
   ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:119:4: note:     function instantiation failed
fn __wrap_and_execute_object_raising_main[
   ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:131:57: note:       call expansion failed - no concrete specializations
    return __wrap_and_execute_raising_main[wrapped_main](argc, argv)
                                                        ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:91:4: note:         function instantiation failed
fn __wrap_and_execute_raising_main[
   ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:107:18: note:           call expansion failed - no concrete specializations
        main_func()
                 ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:128:8: note:             function instantiation failed
    fn wrapped_main() raises:
       ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/builtin/_startup.mojo:129:22: note:               call expansion failed - no concrete specializations
        _ = main_func()
                     ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo:183:5: note:                 function instantiation failed
def main():
    ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo:188:29: note:
  call expansion failed - no concrete specializations
    test_assert_almost_equal()
                            ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo:83:5: note:
  function instantiation failed
def test_assert_almost_equal():
    ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo:142:32: note:
      call expansion failed - no concrete specializations
    _should_fail[DType.bool, 1](True, False)
                               ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo:126:9: note:
       function instantiation failed
    def _should_fail[
        ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/test/testing/test_assertion.mojo:138:32: note:
          call expansion failed - no concrete specializations
            assert_almost_equal(
                               ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/testing/testing.mojo:250:4: note:
   function instantiation failed
fn assert_almost_equal[
   ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/testing/testing.mojo:293:32: note:
      call expansion failed - no concrete specializations
    var almost_equal = _isclose(
                               ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/testing/testing.mojo:36:4: note:
      function instantiation failed
fn _isclose(
   ^
/home/leandro/projects/github.com/leandrolcampos/mojo/stdlib/src/testing/testing.mojo:63:19: note:
         constraint failed: the SIMD type must be numeric
        & (_abs(a - b) <= (atol_vec.max(rtol_vec * _abs(a).max(_abs(b)))))
                  ^
mojo: error: failed to run the pass manager

--

********************
********************
Failed Tests (1):
  Mojo Standard Library :: testing/test_assertion.mojo


Testing Time: 40.30s

Total Discovered Tests: 83
  Unsupported:  8 (9.64%)
  Passed     : 74 (89.16%)
  Failed     :  1 (1.20%)
```

Reply:
I think this might be another instance of #2223, sad.